resolwe/flow/models/entity.py

```diff
@@ -26,3 +27,39 @@ class Meta(BaseCollection.Meta):
 
     #: indicate whether `descriptor` is completed (set by user)
     descriptor_completed = models.BooleanField(default=False)
+
+
+class RelationType(models.Model):
+    name = models.CharField(max_length=100, unique=True)
+
+    ordered = models.BooleanField(default=False)
+
+
+class Relation(BaseModel):
+
+
+    class Meta(BaseCollection.Meta):
+        """Relation Meta options."""
+
+        permissions = (
+            ("view_relation", "Can view relation"),
+            ("edit_relation", "Can edit relation"),
+            ("share_relation", "Can share relation"),
+            ("owner_relation", "Is owner of the relation"),
+        )
+
+    type = models.ForeignKey('RelationType')
+
+    collection = models.ForeignKey('Collection', blank=True)
+
+    entities = models.ManyToManyField(Entity, through='PositionInRelation')
+
+
+class PositionInRelation(models.Model):
+    """Intermediate modell to connect Entities with Relations."""
+
+    entity = models.ForeignKey(Entity, on_delete=models.CASCADE)
+
+    relation = models.ForeignKey(Relation, on_delete=models.CASCADE)
+
+    position = models.CharField(max_length=100, null=True, blank=True)
```

Comment:
I assume the datatype of `position` is based on specific use cases? Perhaps it would be useful to have this as a JSON field in order to be more general (the position can then be an integer or something else).